File: app/record.py
from base import BaseHandler
from base import BASEDIR
from base import authenticated_self
from utils.log import log

from database import db
from utils.mysqlpasswd import mysql_password
import uuid
import json
import tornado.web


class RecordAllHandler(BaseHandler):


    @authenticated_self
    def get(self):
        log.debug("RecordAllHandler get in")
        try:
            index = int(self.get_argument("index"))
        except:
            index = 1
        if index > 0:
            pass
        else:
            index = 1

        data = db.get_record_list(index)
        result = {}
        if data:
            result["result"] = "ok"
            result["maxindex"] = data["maxindex"]
            result["curruntindex"] = data["curruntindex"]
            result["data"] =[]
            for one in data["data"]:
                result["data"].append({"id" : one[0],
                                        "user_id" : one[1],
                                        "object_id" : one[3],
                                        "context": one[5],
                                        "time":one[6]
                                       })

            self.send_data(result)
        else:

            result["result"] = "error"
            result["message"] = "get alarm list failed"
            self.send_data(result)

File: static/js/record.js
/**
 * Created by lenovo on 2016/6/4.
 */
function get_record_list_refresh(index) {

    get_record_list(index, function (data) {
        var result = JSON.parse(data);
        console.log(result);
        if (result.result == "error") {
            console.log("获取日志列表失败")
        }
        else {
            //显示表信息
            $("#record_table tbody").empty();
            for (var i = 0; i < result.data.length; i++) {
                $("#record_table tbody").append("<tr class='default'> <td>" + result.data[i].id +
                    "</td> <td>" + result.data[i].user + "</td> <td>" + result.data[i].object +
                     "</td> <td>" + result.data[i].context+"</td> <td>" + result.data[i].time + "</td> </tr>")
            }

           

            //显示分页条
           split_page("record", result, get_record_list_refresh);
        }

    })

}


